pytensor/gradient.py

```diff
@@ -2052,6 +2052,75 @@ def hessian(cost, wrt, consider_constant=None, disconnected_inputs="raise"):
     return as_list_or_tuple(using_list, using_tuple, hessians)
 
 
+def hessian_vector_product(cost, wrt, p, **grad_kwargs):
+    """Return the expression of the Hessian times a vector p.
+
```

Comment:
Maybe we should mention that this is using backward mode autodiff twice? You can also combine forward and backward autodiff, which might often produce better code I think.

Reply:
The PyTensor docs on derivatives already mention how to build it from the two approaches, only after that do I introduce this helper

Reply:
I'll mention it uses backward mode twice

Reply:
I just meant we could mention it in the docstring? But all good either way.